### PR TITLE
[WIP] Location permission improvements

### PR DESCRIPTION
**Pull request by KennyHuRadar**
- Includes enumerations and logic to handle approximate locations.
- changes internal naming.
- Fixes typos.

### CODE REVIEW

**Comment by KennyHuRadar** [comment]
I had to increase timeouts for some of the test to pass the github CI :(

File: RadarSDK/Include/RadarLocationPermissionStatus.h [hunk 1]
```diff
@@ -20,9 +20,11 @@
 typedef NS_ENUM(NSInteger, RadarLocationPermissionState) {
     NoPermissionGranted,
     ForegroundPermissionGranted,
+    ApproximateForegroundPermissionGranted,
     ForegroundPermissionRejected,
     ForegroundPermissionPending,
     BackgroundPermissionGranted,
+    ApproximateBackgroundPermissionGranted,
```
Comment (ShiCheng-Lu) [inline RadarSDK/Include/RadarLocationPermissionStatus.h:27]: has these been added by apple? if not, then we should leave them out even if android has them.

File: RadarSDK/Include/RadarLocationPermissionStatus.h [hunk 2]
```diff
@@ -18,33 +18,39 @@
...
+    ForegroundAuthorized,
+    ForegroundFullAccuracyDenied,
+    ForegroundAuthorizationDenied,
+    ForegroundAuthorizationRequestInProgress,
+    BackgroundAuthorized,
+    BackgroundFullAccuracyDenied,
+    BackgroundAuthorizationDenied,
+    BackgroundAuthorizationRequestInProgress,
+    AuthorizationRestricted,
     Unknown
```
Comment (KennyHuRadar) [inline RadarSDK/Include/RadarLocationPermissionStatus.h:31]: @lmeier here is my proposed renaming scheme to make our vocabulary more similar to that of Apples.
For reference here is Apple's docs for the new fields: https://developer.apple.com/documentation/corelocation/clservicesessiondiagnostic?changes=latest_minor&language=objc

**Comment by KennyHuRadar** [comment]
@lmeier As per discussed offline:

- We prefer to keep our `foreground` and `background` prefix.
- We will be going with the more colloquial term `permission` vs apple's term of `authorization`
- We will prefer `precise` and maybe `coarse` vs Apple's term `fullAccuracy` 
- We will adopt Apple's phrasing `inProgress` and  `request`
- We will use Apple's term `denied`, which will introduce its counterpart `granted`

**Comment by lmeier** [comment]
@KennyHuRadar looks like you haven't yet pushed here.

**Comment by lmeier** [comment]
Closing for now

**Pull request closed without merging**